### ISSUE TITLE
Adding custom sets isn't working

### ISSUE THREAD

**Issue by BlackKnightJack** [Cockatrice/Cockatrice#3019]
I tried adding a custom set to Cockatrice using Card Database -> Add custom sets/cards and, for whatever reason, it's not working. Currently this is the XML I'm trying to import in its entirety.

```xml
<cockatrice_carddatabase version="3">
  <sets>
    <set>
        <name>CSL</name>
        <longname>Custom Standard League</longname>
        <settype>Expansion</settype>
        <releasedate>2018-02-01</releasedate>
    </set>
  </sets>
  <cards>
    <card>
        <name>Vesama, Empress of Tagnara</name>
        <set rarity="Mythic Rare" picURL="https://i.imgur.com/xf1slRt.png">CSL</set>
        <color>W</color>
        <color>B</color>
        <manacost>3WB</manacost>
        <cmc>5</cmc>
        <type>Legendary Creature - Vampire</type>
        <tablerow>3</tablerow>
        <text>Flying, vigilance {T}: Put a -1/-1 counter on each creature your opponents control. You may activate this ability as if Vesama, Empress of Tagnara had haste.</text>
    </card>
  </cards>
</cockatrice_carddatabase>
```

The strange thing about this, though, is that the client I'm using (2.4.0) was able to apply the recent RIX spoiler patch. Likewise, the set registers just fine when I name it "01.customleague.XML" and put it directly into the custom sets folder, but it won't apply when I use "Add custom sets/cards" regardless of whether the file is called "01.customleague.XML" or "customleague.XML". I have no idea what could be causing this.

**Comment by ZeldaZach** [comment]
Can you attach the XML file here? Drag and drop into a comment

**Comment by Daenyth** [comment]
Does the debug log show anything? What OS and version?

**Comment by ctrlaltca** [comment]
In this case the first step is to validate the xml file as described here: https://github.com/Cockatrice/Cockatrice/wiki/Custom-Cards-&-Sets#i-modified-my-cardsxml-and-it-isnt-working-properly
I already did it for you, and your file validates correctly:
```
fab$ xmllint --schema ../prj/Cockatrice/doc/cards.xsd test.xml 
<?xml version="1.0"?>
<cockatrice_carddatabase version="3">
...
</cockatrice_carddatabase>
test.xml validates
```
I then tried to load the card in Cockatrice using the "Add custom sets/cards" menu entry.
The file has been loaded, the card database reloaded and i've been asked if i wantd to enable the nel set:

![schermata 2018-01-17 alle 18 47 05](https://user-images.githubusercontent.com/1631111/35058065-1554370e-fbb7-11e7-9fb0-b962ca14bc1e.png)

After pressing "yes", the card was available in the card editor:

![schermata 2018-01-17 alle 18 47 38](https://user-images.githubusercontent.com/1631111/35058091-2c7d7ed6-fbb7-11e7-8c41-08010e7b272e.png)



> I have no idea what could be causing this.

Maybe Cockatrice being unable to write the file, eg. a permission issue, or an already existing file with the same name?
